src/main/java/database/service/JsonDatabaseService.java

```diff
@@ -148,6 +148,7 @@ public void removeAllRecordsFromTable(Class<? extends BaseEntity> entityClass) {
         Path databasePath = Path.of(getDatabasePath(entityClass));
         try {
             Files.writeString(databasePath, EMPTY_BRACKETS_TO_JSON);
+            entityIds.clear();
```

Comment:
Please don't clean id counters for **ALL** tables, reset counter only for current table.